LoadFile.go

```diff
@@ -0,0 +1,81 @@
+package crossrefindexer
+
+import (
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+func readFile(path string) error {
```

Comment:
You should add context to the errors returned to give a meaningful error message. Either by `errors.Wrap` or `fmt.Errorf` with `%w`